src/java.prefs/unix/classes/java/util/prefs/FileSystemPreferences.java

```diff
@@ -958,6 +958,8 @@ private boolean lockFile(boolean shared) throws SecurityException{
                 Thread.sleep(sleepTime);
             } catch(InterruptedException e) {
                 checkLockFile0ErrorCode(errorCode);
+                // Don't lose the interrupt unless we throw.
```

Comment:
Why should we lose the interrupted status if we throw a SecurityException? It doesn't look right.

Reply:
I thought that if an access error is found, then that would take precedence over the interrupt, especially since it occurs before the sleep. I was more concerned with just returning false and the caller not knowing if it's false due to interrupt, or false because the file could not be locked after all retries.

Certainly the interrupt status could move before the check which may through, but I think it's less likely that the status will be checked in a catch of SecurityException.

Reply:
The status might not be explicitly checked, but setting the interrupted status will make sure that subsequent calls to sleep/await/tryLock etc. will not block.

In general, we want to preserve the interrupted status until either the user decides that it's fine to clear, or until the thread dies.

Reply:
In which case the code might be simplified to just:
```
            } catch (InterruptedException e) {
                // Don't lose the interrupt
                Thread.currentThread().interrupt();
                break;
            }
```

Reply:
I've reworked the change to always set the interrupted status. I wouldn't remove the check of the error code here, as it'd be a behavior change. I can follow up with that, though it seems to me that it's good to still have the check for access error here.

Reply:
My suggestion was to `break;` instead of `return false;` which would fall through to line 967 where the check method would be called. But what you have is probably preferable as it preserves the original style and keep the changes minimal.